app/demo-geant-integration/demo-geant-integration.cc

```diff
@@ -0,0 +1,128 @@
+#endif
+
+#include "corecel/Assert.hh"
+#include "corecel/io/Logger.hh"
+#include "corecel/sys/Environment.hh"
+#include "celeritas/ext/detail/GeantPhysicsList.hh"
+#include "accel/Logger.hh"
+
+#include "ActionInitialization.hh"
+#include "DetectorConstruction.hh"
+#include "GlobalSetup.hh"
+#include "PrimaryGeneratorAction.hh"
+
+namespace
+{
+//---------------------------------------------------------------------------//
+int GetNumThreads()
+{
+    const std::string& nt_str = celeritas::getenv("NUM_THREADS");
+    if (!nt_str.empty())
+    {
+        try
+        {
+            return std::stoi(nt_str);
+        }
+        catch (const std::logic_error& e)
+        {
+            std::cerr << "error: failed to parse NUM_THREADS='" << nt_str
+                      << "' as an integer" << std::endl;
+            return -1;
+        }
+    }
+    return G4Threading::G4GetNumberOfCores();
+}
+
+//---------------------------------------------------------------------------//
+} // namespace
+
+//---------------------------------------------------------------------------//
+/*!
+ * Execute and run.
+ */
+int main(int argc, char* argv[])
+{
+    std::vector<std::string> args(argv, argv + argc);
+    if (args.size() != 2 || args[1] == "--help" || args[1] == "-h")
+    {
+        std::cerr << "usage: " << args[0] << " {commands}.mac\n"
+                  << "Environment variables:\n"
+                  << "  NUM_THREADS: number of CPU threads\n"
+                  << "  CELER_DISABLE_DEVICE: nonempty disables CUDA\n"
+                  << "  CELER_LOG: global logging level\n"
+                  << "  CELER_LOG_LOCAL: thread-local logging level\n"
+                  << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    // Set up the number of threads
+    CLHEP::HepRandom::setTheSeed(0xcf39c1fa9a6e29bcul);
+
+    int num_threads = GetNumThreads();
+    if (num_threads <= 0)
+    {
+        return EXIT_FAILURE;
+    }
+
+    std::unique_ptr<G4RunManager> run_manager;
+#if CELERITAS_G4_V10
+    if (num_threads > 1)
+    {
+        run_manager = std::make_unique<G4MTRunManager>();
+    }
+    else
+    {
+        run_manager = std::make_unique<G4RunManager>();
+    }
+#else
+    run_manager.reset(G4RunManagerFactory::CreateRunManager(
+        num_threads > 1 ? G4RunManagerType::MT : G4RunManagerType::Serial));
+#endif
+    CELER_ASSERT(run_manager);
+
+    run_manager->SetNumberOfThreads(num_threads);
+    celeritas::self_logger() = celeritas::make_mt_logger(*run_manager);
+
+    celeritas::GeantPhysicsOptions geant_phys_opts{};
+
+    // Construct physics, geometry, celeritas setup, and user setup
+    run_manager->SetUserInitialization(new demo_geant::DetectorConstruction{});
+#if 0
+    // TODO: use full physics
+    run_manager->SetUserInitialization(new FTFP_BERT);
+#else
+    // For now (reduced output) use just EM
+    run_manager->SetUserInitialization(
+        new celeritas::detail::GeantPhysicsList{geant_phys_opts});
+#endif
```

Comment:
This will deal with EM particles only, which may not be a realistic test case at all as only **primary** EM particles will be collected, i.e., there will be no secondary from either hadrons or EM particles on the CPU side.  I would use the default physics list and shoot number of hadrons (either pions or protons) by the particle gun which mimics a simple physics event (if we do not use a set of real physics events for the first pass).